source/upgrading/upgrade/upgrade-4.13.rst

```diff
@@ -0,0 +1,360 @@
+your upgrade procedure, and working through it on a test system before
+working on a production system.
+
+.. note::
+    The following upgrade instructions should be performed regardless of
+    hypervisor type.
+
+Overview of Upgrade Steps:
+----------------------------
+
+#. Check any customisations and integrations
+#. Upload the |sysvm64-version| System VM template if not already using it.
+#. Stop all running management servers
+#. Backup CloudStack database (MySQL)
+#. Upgrade 1st CloudStack management server
+#. Update hypervisors specific dependencies
+#. Restart 1st management server
+#. Check that your upgraded environment works as expected
+#. Upgrade and restart the remaining management servers
+
+
+.. include:: _customisation_warnings.rst
+
+.. warning::
+    If you are not already using the |sysvm64-version| System VM template you will need to 
+    upgrade your System VM template prior to performing the upgrade of the 
+    CloudStack packages.
+
+.. include:: _sysvm_templates.rst
+
+Packages repository
+-------------------
+
+Most users of CloudStack manage the installation and upgrades of
+CloudStack with one of Linux's predominant package systems, RPM or
+APT. This guide assumes you'll be using RPM and Yum (for Red Hat
+Enterprise Linux or CentOS), or APT and Debian packages (for Ubuntu).
+
+Create RPM or Debian packages (as appropriate) and a repository from
+the |release| source, or check the Apache CloudStack downloads page at
+http://cloudstack.apache.org/downloads.html
+for package repositories supplied by community members. You will need
+them for :ref:`ubuntu412` or :ref:`rhel412` and :ref:`kvm412` hosts upgrade.
+
+Instructions for creating packages from the CloudStack source are in the
+`CloudStack Installation Guide`_.
+
+Database Preparation
+--------------------
+
+Backup current database
+
+#. Stop your management server or servers. Run this on all management
+   server hosts:
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-management stop
+
+#. If you are running a usage server or usage servers, stop those as well:
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-usage stop
+
+#. Make a backup of your MySQL database. If you run into any issues or
+   need to roll back the upgrade, this will assist in debugging or
+   restoring your existing environment. You'll be prompted for your
+   password.
+
+   .. parsed-literal::
+
+      $ mysqldump -u root -p cloud > cloud-backup_`date '+%Y-%m-%d'`.sql
+      $ mysqldump -u root -p cloud_usage > cloud_usage-backup_`date '+%Y-%m-%d'`.sql
+
+#. **(KVM Only)** If primary storage of type local storage is in use, the
+   path for this storage needs to be verified to ensure it passes new
+   validation. Check local storage by querying the cloud.storage\_pool
+   table:
+
+   .. parsed-literal::
+
+      $ mysql -u cloud -p -e "select id,name,path from cloud.storage_pool where pool_type='Filesystem'"
```

Comment:
Why can't this be done in the upgrade sql ?

Reply:
@PaulAngus It's boilerplate from older upgrade rst file, I think we may want to review the whole post-upgrade cases and remove/edit as necessary.

Reply:
https://github.com/apache/cloudstack/pull/3813 - check and approve

Reply:
Since PR was merged, I'll remove the workaround ref from the release notes.